File: backend/api/app/Models/LeaveRequest.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class LeaveRequest extends Model
{
    use HasFactory;

    public const STATUS_PENDING = 'pending';
    public const STATUS_APPROVED = 'approved';
    public const STATUS_REJECTED = 'rejected';
    public const STATUS_CANCELLED = 'cancelled';

    protected $fillable = [
        'user_id',
        'type',
        'leave_type_id',
        'date_from',
        'date_to',
        'time_from',
        'time_to',
        'reason',
        'status',
        'doctor_note',
        'permit_photo',
        'approved_by',
        'approved_at',
    ];

    protected $casts = [
        'date_from' => 'date',
        'date_to' => 'date',
        'approved_at' => 'datetime',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function approvedBy()
    {
        return $this->belongsTo(User::class, 'approved_by');
    }

    public function leaveType()
    {
        return $this->belongsTo(LeaveType::class);
    }
}

File: mobile.backup-20251230-070642/lib/models/leave_request.dart
class LeaveRequest {
  const LeaveRequest({
    required this.id,
    required this.userId,
    required this.type,
    required this.dateFrom,
    required this.dateTo,
    required this.reason,
    required this.status,
    this.doctorNote,
    this.createdAt,
    this.updatedAt,
  });

  final int id;
  final int userId;
  final String type;
  final DateTime? dateFrom;
  final DateTime? dateTo;
  final String reason;
  final String status;
  final String? doctorNote;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  factory LeaveRequest.fromJson(Map<String, dynamic> json) {
    return LeaveRequest(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      type: json['type'] as String? ?? '',
      dateFrom: json['date_from'] != null
          ? DateTime.tryParse(json['date_from'] as String)
          : null,
      dateTo: json['date_to'] != null
          ? DateTime.tryParse(json['date_to'] as String)
          : null,
      reason: json['reason'] as String? ?? '',
      status: json['status'] as String? ?? '',
      doctorNote: json['doctor_note'] as String?,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'] as String)
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.tryParse(json['updated_at'] as String)
          : null,
    );
  }
}
